packages/components/pie-button/src/button.scss

```diff
@@ -83,31 +128,37 @@
     }
 
     &[disabled] {
-        --btn-bg-color: var(--dt-color-disabled-01);
-        --btn-text-color: var(--dt-color-content-disabled);
+        --btn-text-color: var(--dt-color-content-disabled) !important;
 
-        border: 1px solid var(--dt-color-disabled-01);
         cursor: not-allowed;
-    }
 
-    &[disabled][variant='ghost'] {
-        --btn-bg-color: var(--dt-color-container-default);
+        // For every variant except ghost, set the disabled background color
+        &:not([variant='ghost']) {
+            --btn-bg-color: var(--dt-color-disabled-01) !important;
+        }
 
-        outline: none;
+        // For outline variants, set the border to the disabled color
+        &[variant='outline'] {
+            border-color: var(--dt-color-disabled-01) !important;
+        }
     }
 
     &[size='xsmall'] {
         --btn-padding: 6px var(--dt-spacing-b);
-        --btn-min-block-size: 32px;
         --btn-font-size: calc(var(--dt-font-size-14) * 1px);
```

Comment:
Feels like a mixin is needed to add px to the font-size and line height tokens. Or should we update pie-design-tokens css and scss builds to add 'px' to those tokens?

Reply:
Yeh, we'll definitely be doing this as part of the CSS tidyup, once we've got the components pretty much complete.

Should be added as a mixin, as we can't add it to the tokens (as apps use `pt` rather and `px`)